packages/libs/components/src/plots/VolcanoPlot.tsx

```diff
@@ -166,12 +203,19 @@ function VolcanoPlot(props: VolcanoPlotProps, ref: Ref<HTMLDivElement>) {
    */
 
   // For the actual volcano plot data
+  // Only return data if the points fall within the specified range! Otherwise they'll show up on the plot.
   const dataAccessors = {
     xAccessor: (d: VolcanoPlotDataPoint) => {
-      return Number(d?.log2foldChange);
+      return Number(d?.log2foldChange) <= xAxisMax &&
+        Number(d?.log2foldChange) >= xAxisMin
+        ? Number(d?.log2foldChange)
+        : null;
     },
     yAccessor: (d: VolcanoPlotDataPoint) => {
-      return -Math.log10(Number(d?.pValue));
+      return -Math.log10(Number(d?.pValue)) <= yAxisMax &&
+        -Math.log10(Number(d?.pValue)) >= yAxisMin
+        ? -Math.log10(Number(d?.pValue))
+        : null;
```

Comment:
Can you create a variable for the result of `-Math.log10(Number(d?.pValue))`, and use that in the comparisons/return?